azure-ts-static-website/staticWebsite.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright 2016-2018, Pulumi Corporation.  All rights reserved.
+
+import * as pulumi from "@pulumi/pulumi";
+
+export interface StorageStaticWebsiteArgs {
+    accountName: pulumi.Input<string>;
+}
+
+const accountNameProp = "accountName";
+
+// There's currently no way to enable the Static Web Site feature of a storage account via ARM
+// Therefore, we created a custom provider which wraps corresponding Azure CLI commands
+class StorageStaticWebsiteProvider implements pulumi.dynamic.ResourceProvider {
+
+    check = async (olds: any, news: any) => {
+        const failedChecks = [];
+
+        if (news[accountNameProp] === undefined) {
+            failedChecks.push({property: accountNameProp, reason: `required property '${accountNameProp}' missing`});
+        }
+
+        return { inputs: news, failedChecks: failedChecks };
+    }
+
+    diff = async (id: pulumi.ID, olds: any, news: any) => {
+        const replaces = [];
+
+        if (olds[accountNameProp] !== news[accountNameProp]) {
+            replaces.push(accountNameProp);
+        }
+
+        return { replaces: replaces };
+    }
+
+    create = async (inputs: any) => {
+        const { execSync } = require("child_process");
+        execSync("az extension add --name storage-preview", { stdio: "ignore" });
```

Comment:
Note that since this is `async` you could fairly easily make this exec call asynchronously.  But I doubt it has any practical performance impact - so fine to keep as-is if it's too hard to make this async.

Reply:
Yeah, I tried, but it wasn't worth the hassle. Apart from promisify thing, there is a difference in options of `exec` vs `execSync` which didn't help.